scene/gui/code_edit.cpp

```diff
@@ -1445,19 +1460,59 @@ bool CodeEdit::is_line_numbers_zero_padded() const {
 }
 
 void CodeEdit::_line_number_draw_callback(int p_line, int p_gutter, const Rect2 &p_region) {
-	String fc = String::num(p_line + 1).lpad(line_number_digits, line_number_padding);
-	if (is_localizing_numeral_system()) {
-		fc = TS->format_number(fc);
-	}
-	Ref<TextLine> tl;
-	tl.instantiate();
-	tl->add_string(fc, theme_cache.font, theme_cache.font_size);
-	int yofs = p_region.position.y + (get_line_height() - tl->get_size().y) / 2;
+	if (!get_rect().intersects(p_region)) {
+		return;
+	}
```

Comment:
This is not correct, if the CodeEdit's rect is not at the origin then line numbers won't appear.

Reply:
i see it now, my bad